code/Matrix/matrix.h

```diff
@@ -0,0 +1,61 @@
+struct Mat {
+  int n, m;
+  vector<vector<int>> a;
+  Mat() { }
+  Mat(int _n, int _m) {n = _n; m = _m; a.assign(n, vector<int>(m, 0)); }
+  Mat(vector< vector<int> > v) { n = v.size(); m = n ? v[0].size() : 0; a = v; }
+  inline void make_unit() {
+    assert(n == m);
+    for (int i = 0; i < n; i++)  {
+      for (int j = 0; j < n; j++) a[i][j] = i == j;
+    }
+  }
+  inline Mat operator + (const Mat &b) {
+    assert(n == b.n && m == b.m);
+    Mat ans = Mat(n, m);
+    for(int i = 0; i < n; i++) {
+      for(int j = 0; j < m; j++) {
+        ans.a[i][j] = (a[i][j] + b.a[i][j]) % mod;
+      }
+    }
+    return ans;
+  }
+  inline Mat operator - (const Mat &b) {
+    assert(n == b.n && m == b.m);
+    Mat ans = Mat(n, m);
+    for(int i = 0; i < n; i++) {
+      for(int j = 0; j < m; j++) {
+        ans.a[i][j] = (a[i][j] - b.a[i][j] + mod) % mod;
+      }
+    }
+    return ans;
+  }
+  inline Mat operator * (const Mat &b) {
+    assert(m == b.n);
+    Mat ans = Mat(n, b.m);
+    for(int i = 0; i < n; i++) {
+      for(int j = 0; j < b.m; j++) {
+        for(int k = 0; k < m; k++) {
+          ans.a[i][j] = (ans.a[i][j] + 1LL * a[i][k] * b.a[k][j] % mod) % mod;
```

Comment:
If you could the curly braces `}`, please ? They are taking 3 lines for nothing.